services/ingestion/etl/agri/csv/src/events/onParseCSV.js

```diff
@@ -2,53 +2,38 @@ import AWS from 'aws-sdk'; // eslint-disable-line import/no-extraneous-dependenc
 import parse from 'csv-parse';
 import transform from 'stream-transform';
 
+// ETL utilities.
+import ensureExtensions from '@eubfr/lib/etl/ensureExtensions';
+import extractMessage from '@eubfr/lib/etl/extractMessage';
+import handleError from '@eubfr/lib/etl/handleError';
+
 import MessengerFactory from '@eubfr/logger-messenger/src/lib/MessengerFactory';
 import { STATUS } from '@eubfr/logger-messenger/src/lib/status';
 
-// Import logic
-import { extractMessage } from '../lib/sns';
 import transformRecord from '../lib/transform';
-import uploadFromStream from '../lib/uploadFromStream';
 
-export const handler = async (event, context, callback) => {
-  // Extract env vars
+export const handler = async (event, context) => {
   const { BUCKET, REGION, STAGE } = process.env;
 
   if (!BUCKET || !REGION || !STAGE) {
-    return callback(
-      new Error('BUCKET, REGION and STAGE environment variables are required!')
+    throw new Error(
+      'BUCKET, REGION and STAGE environment variables are required!'
     );
   }
 
-  // Validate handler execution and check event, context, etc.
   const snsMessage = extractMessage(event);
   const { key } = snsMessage.object;
 
-  const messenger = MessengerFactory.Create({ context });
-
-  const handleError = async (e, cb) => {
-    await messenger.send({
-      message: {
-        computed_key: key,
-        status_message: e.message,
-        status_code: STATUS.ERROR,
-      },
-      to: ['logs'],
-    });
-
-    return cb(e);
-  };
+  if (!ensureExtensions({ file: key, extensions: ['.csv'] })) {
+    throw new Error('CSV file expected for this ETL.');
```

Comment:
I honestly don't think so, here's my way of thinking.

`handleError` is a utility which is used to communicate errors to users via the `messanger`, it's true it's used throughout when there would be possibly errors related to parsing, transformations, upload or anything else which the user can actually fix. (by fixing the data and re-uploading the file)

The file extensions check is a redundant in my opinion, but even when I did 57a195a1003665a044b9b579bb438eb7fddec944 I left the checks to be in the event handlers, mostly to not make too big of a change for someone remembering the old way of the ETLs.

The reason these checks are reduntant is because the ingestion manager will try to reach an SNS topic [endpoint based on the extension uploaded by a given producer](https://github.com/ec-europa/eubfr-data-lake/blob/master/services/ingestion/manager/src/events/onObjectCreated.js#L57). So, if a user from AGRI uploads an `.xls` file, for example, the ingestion manager will try to reach a topic, which does not exist, and thus the `onParseCSV.js` will never get invoked.

The user will get feedback about [this case from the ingestion manager](https://github.com/ec-europa/eubfr-data-lake/blob/master/services/ingestion/manager/src/events/onObjectCreated.js#L166).

Reply:
Oh I see! Thanks for the explanation! 😀